src/internal/m/m.go

```diff
@@ -0,0 +1,441 @@
+// Package m implements lightweight metrics.
+
+// A metric is a name, a set of key/value pairs (set by
+// the underlying logger), and a value.  A value can be an absolute value ("gauge"), an incremental
+// value ("counter"), or a set of samples (like "histogram", though lossless in this
+// implementation).  Metrics are most useful when additional code analyzes the entire log of a
+// program's run; this is called "analysis code" or "anaylsis software" throughout the
+// documentation.
+//
+// Values are stored by writing the set of operations to build the value to the logs.  For example,
+// each time a sample is added to a sampler with Sample, a log line is produced.  Reading the logs
+// will allow analysis software to recover the entire set of samples.  Counters are similar; each
+// increment event emits a log line, and and analysis code can add the deltas to see the final
+// value, or the value at a particular point in time.
+//
+// Values are logically associated with key=value metadata.  Each value is uniquely identified by
+// the set of key=value metadata; a metric foo with field foo=bar is logically a different value
+// from the metric foo with field foo=baz.  The metadata is defined by the fields applied to the
+// underlying logger.  For example, if you declare a counter `tx_bytes` on each HTTP request you
+// handle, each time the count of transmitted bytes is updated, the log message will include fields
+// inherited from the default HTTP logger, which includes a per-request ID.  Therefore, analysis
+// code can calculate a tx_byte count for every request handled by the system by observing the value
+// of the x-request-id field on log lines matching the tx_byte counter format.  And, of course, it
+// can ignore the extra fields and add everything up to show a program-wide count of bytes
+// transmitted.
+//
```

Comment:
Good point.  I added documentation about not needing to ambiguate this because pctx.Child is the place to do it.  I also fleshed out the documentation on Child to describe how it works, and what naming conventions should be used there.